### PR TITLE
fix: projected AwkwardInputLayers have __dask_tokenize__

### PR DESCRIPTION
**Pull request by lgray**
Another possible fix to #468, perhaps the simplest.

@agoose77 @martindurant @jpivarski

### CODE REVIEW

File: src/dask_awkward/layers/layers.py [hunk 1]
```diff
@@ -117,6 +117,19 @@ def io_func_implements_report(func: ImplementsIOFunction) -> bool:
...
 
 
+class AwkwardTokenizable:
+
+    def __init__(self, ret_func, parent_name):
+        self.parent_name = parent_name
+        self.ret_func = ret_func
+
+    def __dask_tokenize__(self):
+        return (AwkwardTokenizable, self.parent_name)
```
Comment (martindurant) [inline src/dask_awkward/layers/layers.py:127]: May as well 
```suggestion
        return ("AwkwardTokenizable", self.parent_name)
```
or other literal we make just once

File: src/dask_awkward/layers/layers.py [hunk 2]
```diff
@@ -117,6 +117,19 @@ def io_func_implements_report(func: ImplementsIOFunction) -> bool:
...
+
+    def __call__(self, *_, **__):
+        return self.ret_func
```
Comment (martindurant) [inline src/dask_awkward/layers/layers.py:130]:
```suggestion
        return self.ret_func()
```
?

Comment (lgray) [reply src/dask_awkward/layers/layers.py:130]: the original code says otherwise `io_func=lambda *_, **__: new_return,`

Comment (lgray) [reply src/dask_awkward/layers/layers.py:130]: right that is a value not a function :D

**Pull request merged**